### DISCUSSION TITLE
How do Sidecar Metadata Files Work in Komga?

### DISCUSSION THREAD

**Discussion by sidney-eliot** [gotson/komga#1794] [Q&A] [answered]
Would it be possible to have metadata of either an entire comic folder or a single comic issue in a sidecar metadata file? Like ComicInfo.xml, or some other metadata container like [OPF](https://idpf.org/epub/20/spec/OPF_2.0_final_spec.html) or [OPS](https://idpf.org/epub/20/spec/OPS_2.0.1_draft.htm).

Reasoning for not wanting the metadata in the CBZ file:

Because I want it to be easily editable without having to re-zip the comic. And I also don't want to add it via the Komga UI, as that can easily be messed up if the file is renamed in the file explorer, for example or x other reason.

---

An example of what I'm talking about:

(single metadata file for entire comic)

📂Spider Man
└─ Issue 1.cbz
└─ Issue 2.cbz
└─ Issue 3.cbz
└─ ComicInfo.xml

(one metadata file for every issue)

📂Spider Man
└─ Issue 1.cbz
└─ Issue 2.cbz
└─ Issue 3.cbz
└─ Issue 1.xml
└─ Issue 2.xml
└─ Issue 3.xml

It seems like as soon as I zip the xml file, even if nothing else is in the zip, Komga can read it, which seems odd. Also worth noting, is that renaming a regular folder metadata.cbz does not work in tricking Komga.

Cheers,
Sidney

**Comment by gotson** [comment, marked as answer]
This is not part of the accepted standard for ComicInfo.xml. No application producing ComicInfo.xml actually do that.

related to https://github.com/gotson/komga/discussions/1713

> Because I want it to be easily editable without having to re-zip the comic

You should not edit those files manually anyway, better use something like ComicTagger or MangaTagger.

> And I also don't want to add it via the Komga UI, as that can easily be messed up if the file is renamed in the file explorer, for example or x other reason.

I don't quite understand this statement. Would you be able to supplement?

**Reply by sidney-eliot** [reply]
Ok, I'm sold, Manga Manager is great. One question, though, regarding collections.

For this demonstration, I have 3 series:

_The arrows (>>>) indicate what I tagged the issues as in Manga Manager._

📂Test Folder
└─ Issue 1 A.cbz >>> Series = sA, Series Group = Test
└─ Issue 2 AA.cbz >>> Series = sA, Series Group = Test
└─ Issue 3 A.AAcbz >>> Series = sA, Series Group = Test
└─ Issue 1 B.cbz >>> Series = sB, Series Group = Test
└─ Issue 2 BB.cbz >>> Series = sB, Series Group = Test
└─ Issue 1 C.cbz >>> Series = sC, Series Group = Test
└─ Issue 2 CC.cbz >>> Series = sC, Series Group = Test


Then when looking into the collection view of Komga, I see:

**Test** (which is as expected)

Clicking the **Test** collection, reveals:

**sA** (what I expected to see was sA, sB, sC)

And clicking the **sA** series, reveals all the issues from all the three series.

---

Interestingly, adding all series into their own folder, like so:

📂Test Folder
└─📂 sA
└── Issue 1 A.cbz >>> Series = sA, Series Group = Test
└── Issue 2 AA.cbz >>> Series = sA, Series Group = Test
└── Issue 3 A.AAcbz >>> Series = sA, Series Group = Test
└─📂 sB
└──Issue 1 B.cbz >>> Series = sB, Series Group = Test
└── Issue 2 BB.cbz >>> Series = sB, Series Group = Test
└─📂 sC
└── Issue 1 C.cbz >>> Series = sC, Series Group = Test
└── Issue 2 CC.cbz >>> Series = sC, Series Group = Test

Will properly show the series as their own in the collection view instead of putting them all under series A. Is this some sort of bug? It would be great to not have to re-arange all series into their own folders.



P.S. I really like how every issue shows the collection it's inside of

**Reply by gotson** [reply]
Collections collect series, if you want to collect individual books you need to use Read Lists in Komga.

**Reply by gotson** [reply]
Each series in Komga needs to be its own folder, you can't mix books from different series in one folder, and expects the metadata in ComicInfo.xml to arrange things neatly, that's not how Komga works.

**Reply by sidney-eliot** [reply]
> Each series in Komga needs to be its own folder, you can't mix books from different series in one folder, and expects the metadata in ComicInfo.xml to arrange things neatly, that's not how Komga works.

Don't you often say that Komga mostly disregards folder structure (stuff like deeper folders) and prioritizes metadata. It's just kind of odd that in this case the series metadata is disregarded and the folder structure is used instead.

Obviously if one were to do changes like this, one would have to take into regard if anyone's existing Komga setup might be influenced by it in a bad way, so the call is up to you, but maybe consider this:

Currently, the series metadata field is not considered inside the collection view and only used in the reading list view. So basically, the user tags the issues with their series names and then, in addition to that, also has to set up the folders. Shouldn't the user be able to use one of the two methods and not have to use both at the same time, it just seems redundant.

Cheers

**Reply by gotson** [reply]
Komga requires 1 folder per series, since forever. That's how 99% of people sort their books on disk anyway. If that doesn't work for you, you may consider other software.

The series name in metadata is used to change the series name, check how metadata is used here: https://komga.org/docs/guides/scan-analysis-refresh/#import-metadata-for-cbrcbz-containing-a-comicinfoxml-file

**Reply by sidney-eliot** [reply]
> Komga requires 1 folder per series, since forever ...

Ok, got it. One last question on this topic. How would one handle PDFs. From what I understand, there isn't really any established metadata for them. 

Putting them in Series folders that are in a Series Group folder also doesn't work as Komga needs for the files to be tagged with their series group name, and won't create the collection otherwise.

Maybe instead of allowing metadata to create collections via series groups metadata, one could allow the group series folder and series folders (which as you say 99% already use) to create collections in Komga.

P.S. Thanks for the quick replies.

**Reply by sidney-eliot** [reply]
I also just noticed that the one shot directory works well with collections, so I won't have to put every oneshot in their own folder. 👍

**Reply by gotson** [reply]
> How would one handle PDFs. From what I understand, there isn't really any established metadata for them.

In Komga directly.

**Reply by sidney-eliot** [reply]
> In Komga directly.

Fair, PDF is a hard file format to work with in this context, I think the better approach is for me to simply convert my PDFs (as long as they aren't books) to CBZ files.

I see now that loose ComicInfo.xml files all over the place is not a good idea.
